src/frontend/src/session.rs

```diff
@@ -932,17 +972,18 @@ impl Session for SessionImpl {
         stmt: Statement,
         format: Format,
     ) -> std::result::Result<PgResponse<PgResponseStream>, BoxedError> {
-        let sql_str = stmt.to_string();
+        let sql = Arc::new(stmt.to_string());
         let rsp = {
-            let mut handle_fut = Box::pin(handle(self, stmt, &sql_str, vec![format]));
+            let mut handle_fut = Box::pin(handle(self, stmt, sql.clone(), vec![format]));
             if cfg!(debug_assertions) {
                 // Report the SQL in the log periodically if the query is slow.
                 const SLOW_QUERY_LOG_PERIOD: Duration = Duration::from_secs(60);
                 loop {
                     match tokio::time::timeout(SLOW_QUERY_LOG_PERIOD, &mut handle_fut).await {
                         Ok(result) => break result,
                         Err(_) => tracing::warn!(
-                            sql_str,
+                            "{} {}",
+                            &sql,
```

Comment:
It would be better to keep `sql` a field of the structured log, instead of interpolating it into the message.